modules/twinkleblock.js

```diff
@@ -982,13 +982,14 @@ Twinkle.block.prev_reason = null;
 
 Twinkle.block.callback.preview = function twinkleblockcallbackPreview(form) {
 	var params = {
-		template: form.template.value,
 		article: form.article.value,
-		expiry: form.template_expiry ? form.template_expiry.value : form.expiry.value,
-		indefinite: !!(/indef|infinity|never|\*|max/).exec( form.template_expiry ? form.template_expiry.value : form.expiry.value ),
 		blank_duration: form.blank_duration ? form.blank_duration.checked : false,
 		disabletalk: form.disabletalk.checked || (form.notalk ? form.notalk.checked : false),
-		reason: form.block_reason.value
+		expiry: form.template_expiry ? form.template_expiry.value : form.expiry.value,
+		hardblock: Twinkle.block.isRegistered ? form.autoblock.checked : form.hardblock.checked,
+		indefinite: !!(/indef|infinity|never|\*|max/).exec( form.template_expiry ? form.template_expiry.value : form.expiry.value ),
```

Comment:
If you want just `true`/`false` you could just use `.test` instead and omit the `!!`.